File: scripts/capture-local-arcade.py
"""Current local Arcade screenshots for the owner's visual review; no credentials."""

import json
from pathlib import Path

from playwright.sync_api import sync_playwright

output = Path(__file__).resolve().parents[1] / "docs/evidence/screenshots/arcade-review-20260908"
output.mkdir(parents=True, exist_ok=True)

with sync_playwright() as playwright:
    browser = playwright.chromium.connect_over_cdp("http://127.0.0.1:9327")
    page = browser.contexts[0].new_page()
    page.set_default_timeout(15000)
    page.set_default_navigation_timeout(30000)
    for label, width, height in (("desktop", 1440, 1000), ("mobile", 390, 844)):
        page.set_viewport_size({"width": width, "height": height})
        for surface, path in (("hub", "/practice"), ("drill", "/practice?set=next-move")):
            page.goto("http://learner.localhost:3100" + path, wait_until="domcontentloaded")
            page.wait_for_timeout(3000)
            page.screenshot(path=str(output / f"{surface}-{label}.png"), full_page=True)
    page.set_viewport_size({"width": 1440, "height": 1000})
    page.goto("http://learner.localhost:3100/practice", wait_until="networkidle")
    print(json.dumps({"screenshots": 4, "directory": str(output)}))

File: scripts/capture-local-arcade.mjs
// Page-target CDP avoids browser-wide attachment to Chrome's browser_ui targets.
import { mkdir, writeFile } from "node:fs/promises";
import path from "node:path";

const targets = await fetch("http://127.0.0.1:9327/json/list").then((r) => r.json());
const target = targets.find((t) => t.type === "page" && new URL(t.url).hostname === "learner.localhost");
if (!target) throw new Error("Authenticated local learner tab unavailable");
const socket = new WebSocket(target.webSocketDebuggerUrl);
await new Promise((resolve, reject) => { socket.addEventListener("open", resolve, { once: true }); socket.addEventListener("error", reject, { once: true }); });
let nextId = 0;
const pending = new Map();
socket.addEventListener("message", ({ data }) => {
  const result = JSON.parse(data);
  if (pending.has(result.id)) {
    const { resolve, reject, timer } = pending.get(result.id);
    pending.delete(result.id); clearTimeout(timer);
    if (result.error) reject(new Error("Local page command failed")); else resolve(result.result);
  }
});
const send = (method, params = {}) => new Promise((resolve, reject) => {
  const id = ++nextId;
  const timer = setTimeout(() => { pending.delete(id); reject(new Error("Local page command timed out")); }, 15000);
  pending.set(id, { resolve, reject, timer });
  socket.send(JSON.stringify({ id, method, params }));
});
const output = path.resolve("docs/evidence/screenshots/arcade-review-20260908");
await mkdir(output, { recursive: true });
try {
  await send("Page.enable");
  for (const [label, width, height] of [["desktop", 1440, 1000], ["mobile", 390, 844]]) {
    await send("Emulation.setDeviceMetricsOverride", { width, height, deviceScaleFactor: 1, mobile: false });
    for (const [surface, suffix] of [["hub", "/practice"], ["drill", "/practice?set=next-move"]]) {
      await send("Page.navigate", { url: "http://learner.localhost:3100" + suffix });
      await new Promise((resolve) => setTimeout(resolve, 5000));
      const screenshot = await send("Page.captureScreenshot", { format: "png", captureBeyondViewport: true });
      await writeFile(path.join(output, `${surface}-${label}.png`), Buffer.from(screenshot.data, "base64"));
      process.stdout.write(`${surface}-${label}.png saved\n`);
    }
  }
  await send("Emulation.clearDeviceMetricsOverride");
} finally { socket.close(); }
